wsgi_kerberos.py

```diff
@@ -202,4 +209,8 @@ def custom_start_response(status, headers, exc_info=None):
             return self._unauthorized(start_response, server_token)
         else:
             # Otherwise, return a 403.
-            return self._forbidden(environ, start_response)
+            return self._forbidden(
+                environ,
+                start_response,
+                extra_body=b'\n\n%r' % exc
```

Comment:
Note: I don't love how this composes with the existing support for custom response bodies[1], but maybe it's good enough.

[1] Rather than accepting a string or tuple in the constructor for a custom body and content-type, normalizing, and then not knowing here whether we're changing something custom that the user wants to do, the constructor could instead accept a callback for creating a custom body that we pass the extra info to, and the user can use or ignore it as they see fit. (And I'd accept that as a separate param from the custom content type, rather than having to keep packing/unpacking them into/from the 2-tuple.) But think it's better to optimize for debuggability of error-prone failure modes (e.g. forgetting to init keytabs) over API redesign for more customizable error pages.

Reply:
I don’t think it’s a good idea to leak this information back to the client in the body by default.

Reply:
In the latest revision, the exception is only sent back to the client when we were initialized with `debug=True`.

Reply:
> I don’t think it’s a good idea to leak this information back to the client in the body by default.

Based on various parts of the HTTP specs, I think this sort of thing is okay. For example, https://tools.ietf.org/html/rfc7231#section-6.5.3 says, "A server that wishes to make public why the request has been forbidden can describe that reason in the response payload (if any)." And https://tools.ietf.org/html/rfc7235#section-3.1 says, upon getting a 401 response, "the user agent SHOULD present the enclosed representation to the user, since it usually contains relevant diagnostic information."